lib/jellygrinder/client/helpers/utils.ex

```diff
@@ -63,6 +63,8 @@ defmodule Jellygrinder.Client.Helpers.Utils do
            }, data}
 
         {:error, reason} ->
+          handle_failed_request(reason, state)
```

Comment:
Yes, in the current implementation, we reconnect if any request failed with the reason `:closed`, and still note that the request failed. The reconnect is so that any future requests don't fail